### PR TITLE
[0.12]  	Add backwards-compatibility issues (chainstate obfuscation) to release notes

### PR DESCRIPTION
**Pull request by maflcko**

### CODE REVIEW

**Comment by luke-jr** [comment]
Concept ACK. Is there a way to instruct 0.12 to not use obfuscation?

File: doc/release-notes.md [hunk 1]
```diff
@@ -40,8 +42,17 @@ bootstrap.dat) anew afterwards. It is possible that the data from a completely
...
-known problems when downgrading from 0.11.x to 0.10.x.
+This does not affect wallet forward or backward compatibility.
+
+# Downgrade to a version < 0.12.0
+
+Because release 0.12.0 and later will obfuscate the chainstate on every
+reindex, the block files and databases are not backwards-compatible with
+pre-0.12 versions of Bitcoin Core or other software.
+
+If you want to downgrade after you did a reindex with 0.12.0 or later,
```
Comment (fanquake) [inline doc/release-notes.md]: s/you did/you have done

File: doc/release-notes.md [hunk 2]
```diff
@@ -40,8 +42,17 @@ bootstrap.dat) anew afterwards. It is possible that the data from a completely
...
+you need to reindex when you first start Bitcoin Core version 0.11 or
```
Comment (fanquake) [inline doc/release-notes.md]: s/you/you'll

**Comment by MarcoFalke** [comment]
Addressed precious nits by @morcos and @fanquake

**Comment by laanwj** [comment]
Thanks, makes sense!

**Pull request merged**